### PR TITLE
Make sure that the search scheduler doesn't run to many tasks in parallel

### PR DESCRIPTION
**Pull request by Fox32**
Change search scheduler from starting indexing in a fixed interval (for example
every 60 seconds), to wait a fixed time between index runs.
This makes sure that no second index process for the same document type is
started when the previous one is still running.

We sometimes have the issue that the indexing takes longer than our interval and therefore more and more indexing processes are piling up.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search-backend-node/src/runPeriodically.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
```
Comment (Fox32) [inline plugins/search-backend-node/src/runPeriodically.ts:1]: I borrowed this code from https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/util/runPeriodically.ts
But I don't think that is makes sense to put in into a shared place at it isn't used in the new catalog anymore?

**Review by iamEAP — COMMENTED** [review]
Nice!  Love the incremental improvement.  I was thinking maybe there needed to be an update to a JSDoc string somewhere, but I think it was sufficiently ambiguous to begin with such that no update is needed.

Could you add a little test that checks that a task that takes longer than its interval isn't re-invoked after the interval?

File: plugins/search-backend-node/src/Scheduler.ts [hunk 2]
```diff
@@ -56,11 +57,7 @@ export class Scheduler {
     this.schedule.forEach(({ task, interval }) => {
       // Fire the task immediately, then schedule it.
       task();
```
Comment (freben) [inline plugins/search-backend-node/src/Scheduler.ts:59]: Hm so ... this one isn't awaited. And then an overlapping one will be started immediately, right? Feels like we could skip this line as long as `runPeriodically` does fire right away

Comment (Fox32) [reply plugins/search-backend-node/src/Scheduler.ts:59]: Ohhh, you are so correct! Missed to delete that line when taking it over from my fork...

**Comment by Fox32** [comment]
> Nice! Love the incremental improvement. I was thinking maybe there needed to be an update to a JSDoc string somewhere, but I think it was [sufficiently ambiguous](https://github.com/backstage/backstage/blob/master/plugins/search-backend-node/src/types.ts#L32) to begin with such that no update is needed.
> 
> Could you add a little test that checks that a task that takes longer than its interval isn't re-invoked after the interval?

@iamEAP Done!

**Review by iamEAP — APPROVED** [review]
Nice!

**Comment by Fox32** [comment]
Arg, missed to choose the right name for the changeset, so I need another approval too 🙄

**Comment by Fox32** [comment]
Accidentally pushed some commit into this branch… but I removed them again.

**Review by freben — APPROVED** [review]

**Pull request merged**